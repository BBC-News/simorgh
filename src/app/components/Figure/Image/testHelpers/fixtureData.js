export const landscape = {
  alt: 'Student sitting an exam',
  src:
    'https://ichef.bbci.co.uk/news/1024/cpsprodpb/7098/production/_104842882_students.jpg',
  width: 1024,
  height: 576,
};

export const portrait = {
  alt:
    'Map of the UK displaying Syrian refugees and asylum seekers per 10000 population. Ranges from 0 to 17.',
  src:
    'https://ichef.bbci.co.uk/news/640/cpsprodpb/439A/production/_100960371_syrians_and_asylum_v2-nc.png',
  width: 640,
  height: 1138,
};

export const square = {
  alt: 'Tracks through the snow',
  src:
    'https://ichef.bbci.co.uk/news/1024/cpsprodpb/114FE/production/_104801907_79010.jpg',
  width: 1024,
  height: 1024,
};

export const custom = {
  alt: 'By Elisa Decker, from her series "Sidewalk"',
  src:
    'https://ichef.bbci.co.uk/news/445/cpsprodpb/164BB/production/_104032319_03270dcc-9dda-4bd4-96a0-db89f6b915ae.jpg',

  width: 445,
  height: 547,
};
