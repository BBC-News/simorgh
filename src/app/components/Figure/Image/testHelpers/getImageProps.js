function getImageProps(image, includeHeight) {
  const props = {
    alt: image.alt,
    src: image.src,
    width: image.width,
  };
  props.height = includeHeight ? image.height : null;
  return props;
}

export default getImageProps;
