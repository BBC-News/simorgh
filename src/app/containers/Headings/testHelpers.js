const blocksSingleFragment = (text, attributes = []) => [
  {
    id: '7987213',
    type: 'text',
    model: {
      blocks: [
        {
          id: '6897239',
          type: 'paragraph',
          model: {
            text,
            blocks: [
              {
                id: '4209872',
                type: 'fragment',
                model: {
                  text,
                  attributes,
                },
              },
            ],
          },
        },
      ],
    },
  },
];

export default blocksSingleFragment;
