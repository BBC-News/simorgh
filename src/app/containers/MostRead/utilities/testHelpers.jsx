import React from 'react';
import MostReadContainer from '..';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';

export const setStalePromoTimestamp = mostReadData => {
  const oldTimestamp = 864691200; // 27/05/1997
  const updatedMostReadData = mostReadData;

  // set first promo to have an old timestamp
  updatedMostReadData.records[0].promo.timestamp = oldTimestamp;

  return updatedMostReadData;
};

export const setFreshPromoTimestamp = mostReadData => {
  const freshDate = new Date();
  const updatedMostReadData = mostReadData;

  // Updates first 10 promos to have a fresh date
  for (let i = 0; i < 10; i += 1) {
    updatedMostReadData.records[i].promo.timestamp = freshDate.getTime();
  }

  return updatedMostReadData;
};

export const setStaleLastRecordTimeStamp = mostReadData => {
  const updatedMostReadData = mostReadData;
  updatedMostReadData.lastRecordTimeStamp = '2019-11-06T16:28:00Z';

  return updatedMostReadData;
};

const getToggleState = enabled => ({
  local: { mostRead: { enabled } },
  test: { mostRead: { enabled } },
});

/* eslint-disable react/prop-types */
export const MostReadWithContext = ({
  isAmp = false,
  service,
  variant = null,
  mostReadToggle = false,
}) => (
  <ToggleContext.Provider
    value={{ toggleState: getToggleState(mostReadToggle) }}
  >
    <RequestContextProvider
      bbcOrigin={`http://localhost:7080/${service}`}
      isAmp={isAmp}
      pageType="frontPage"
      service={service}
      statusCode={200}
      pathname={`/${service}`}
      variant={variant}
    >
      <ServiceContextProvider service={service} variant={variant}>
        <MostReadContainer />
      </ServiceContextProvider>
    </RequestContextProvider>
  </ToggleContext.Provider>
);
/* eslint-enable react/prop-types */
