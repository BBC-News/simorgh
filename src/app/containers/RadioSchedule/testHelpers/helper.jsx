/* eslint-disable react/prop-types */
import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';
import { boolean } from '@storybook/addon-knobs';
import React from 'react';
import ProgramCard from '../ProgramCard';
import { ServiceContextProvider } from '#contexts/ServiceContext';

// Will remove and clean up in future PRs
export const stateTypes = ['live', 'onDemand', 'onDemand', 'next'];
export const uniqueStates = ['live', 'onDemand', 'next'];

export const renderProgramCard = ({
  state,
  script,
  service = 'news',
  duration = 'PT30M',
  durationLabel = 'Duration %duration%',
  startTime = 1566914061212,
  displaySummary = boolean('show summary', true),
  linkComponent = 'a',
  linkComponentAttr = 'href',
}) => {
  const { text, articlePath, longText, dir } = TEXT_VARIANTS[service];

  const props = {
    durationLabel: dir === 'rtl' ? 'المدة الزمنية %duration%' : durationLabel,
    linkComponent,
    linkComponentAttr,
  };

  const program = {
    state,
    link: articlePath,
    startTime,
    brandTitle: text,
    summary: displaySummary ? longText : null,
    duration,
  };

  return (
    <ServiceContextProvider
      script={script}
      service={service}
      locale="en-gb"
      dir={dir}
      timezone="Europe/London"
    >
      <ProgramCard program={program} {...props} />
    </ServiceContextProvider>
  );
};
