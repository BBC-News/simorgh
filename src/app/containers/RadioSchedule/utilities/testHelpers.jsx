import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import RadioScheduleContainer from '..';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';

const getToggleState = enabled => ({
  radioSchedule: { enabled },
});

/* eslint-disable react/prop-types */
const RadioSchedulesWithContext = ({
  service,
  radioScheduleToggle = false,
}) => (
  <ToggleContext.Provider
    value={{
      toggleState: getToggleState(radioScheduleToggle),
      toggleDispatch: jest.fn(),
    }}
  >
    <RequestContextProvider
      isAmp={false}
      pageType="frontPage"
      service={service}
      pathname={`/${service}`}
      timeOnServer={Date.now()}
    >
      <ServiceContextProvider service={service}>
        <BrowserRouter>
          <RadioScheduleContainer />
        </BrowserRouter>
      </ServiceContextProvider>
    </RequestContextProvider>
  </ToggleContext.Provider>
);
/* eslint-enable react/prop-types */

export default RadioSchedulesWithContext;
