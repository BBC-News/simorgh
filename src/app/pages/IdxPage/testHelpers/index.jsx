import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import IdxPage from '#pages/IdxPage';

const radioServiceOverride = 'dari';

// eslint-disable-next-line react/prop-types
const IdxPageWithContext = ({ service = 'persian', pageData }) => (
  <BrowserRouter>
    <ToggleContextProvider service={service} origin="https://www.test.bbc.com">
      <RequestContextProvider
        pageType="IDX"
        service={service}
        pathname="/pathname"
        data={{ status: 200 }}
        isAmp={false}
      >
        <ServiceContextProvider service={service}>
          <IdxPage
            pageData={pageData}
            radioScheduleEndpointOverride={`./data/${service}/bbc_${radioServiceOverride}_radio/schedule.json`}
          />
        </ServiceContextProvider>
      </RequestContextProvider>
    </ToggleContextProvider>
  </BrowserRouter>
);

export default IdxPageWithContext;
