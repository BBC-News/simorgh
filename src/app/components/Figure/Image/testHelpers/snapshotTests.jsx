import React from 'react';
import { shouldMatchSnapshot } from '../../../../helpers/tests/testHelpers';
import { custom, landscape, portrait, square } from './fixtureData';

const snapshotTests = Component => {
  shouldMatchSnapshot(
    'should render landscape image correctly',
    <Component
      alt={landscape.alt}
      src={landscape.src}
      height={landscape.height}
      width={landscape.width}
    />,
  );
  shouldMatchSnapshot(
    'should render portrait image correctly',
    <Component
      alt={portrait.alt}
      src={portrait.src}
      height={portrait.height}
      width={portrait.width}
    />,
  );
  shouldMatchSnapshot(
    'should render square image correctly',
    <Component
      alt={square.alt}
      src={square.src}
      height={square.height}
      width={square.width}
    />,
  );
  shouldMatchSnapshot(
    'should render image with custom dimensions correctly',
    <Component
      alt={custom.alt}
      src={custom.src}
      height={custom.height}
      width={custom.width}
    />,
  );
  shouldMatchSnapshot(
    'should render image with srcset correctly',
    <Component
      alt={landscape.alt}
      src={landscape.src}
      srcset={landscape.srcset}
      height={landscape.height}
      width={landscape.width}
    />,
  );
};

export default snapshotTests;
