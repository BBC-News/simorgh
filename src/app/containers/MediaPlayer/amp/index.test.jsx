import React from 'react';
import { shouldMatchSnapshot } from '../../../../testHelpers';
import AmpMediaPlayer from '.';

describe('Media Player: AmpMediaPlayer', () => {
  shouldMatchSnapshot(
    'should render an amp-iframe with placeholder',
    <AmpMediaPlayer
      embedSrc="https://foo.bar/"
      placeholderSrc="https://foo.bar/placeholder.png"
    />,
  );
});
