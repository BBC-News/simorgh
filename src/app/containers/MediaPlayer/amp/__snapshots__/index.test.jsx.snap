// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Player: AmpMediaPlayer should render an amp-iframe with placeholder 1`] = `
<amp-iframe
  allowfullscreen={true}
  frameborder="0"
  layout="fill"
  sandbox="allow-scripts allow-same-origin"
  src="https://foo.bar/"
>
  <amp-img
    layout="fill"
    placeholder={true}
    src="https://foo.bar/placeholder.png"
  />
</amp-iframe>
`;
